extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js

```diff
@@ -326,28 +325,30 @@ export class AmpLightboxGallery extends AMP.BaseElement {
    * @private
    */
   buildDescriptionBox_() {
-    this.descriptionBox_ = this.win.document.createElement('div');
-    this.descriptionBox_.classList.add('i-amphtml-lbg-desc-box');
-    this.descriptionBox_.classList.add('i-amphtml-lbg-controls');
-    this.descriptionBox_.classList.add('standard');
+    this.vsync_.mutate(() => {
+      this.descriptionBox_ = this.win.document.createElement('div');
+      this.descriptionBox_.classList.add('i-amphtml-lbg-desc-box');
+      this.descriptionBox_.classList.add('i-amphtml-lbg-controls');
+      this.descriptionBox_.classList.add('standard');
 
-    // public css api for styling the description box
-    this.descriptionBox_.classList.add('amp-lbg-desc-box');
+      // public css api for styling the description box
+      this.descriptionBox_.classList.add('amp-lbg-desc-box');
 
-    this.descriptionTextArea_ = this.win.document.createElement('div');
-    this.descriptionTextArea_.classList.add('i-amphtml-lbg-desc-text');
-    this.descriptionTextArea_.classList.add('non-expanded');
+      this.descriptionTextArea_ = this.win.document.createElement('div');
+      this.descriptionTextArea_.classList.add('i-amphtml-lbg-desc-text');
+      this.descriptionTextArea_.classList.add('non-expanded');
 
-    // public css api for styling the description text area
-    this.descriptionTextArea_.classList.add('amp-lbg-desc-text');
+      // public css api for styling the description text area
+      this.descriptionTextArea_.classList.add('amp-lbg-desc-text');
 
-    this.descriptionBox_.appendChild(this.descriptionTextArea_);
+      this.descriptionBox_.appendChild(this.descriptionTextArea_);
 
-    this.descriptionBox_.addEventListener('click', event => {
-      this.toggleDescriptionOverflow_();
-      event.stopPropagation();
+      this.descriptionBox_.addEventListener('click', event => {
+        this.toggleDescriptionOverflow_();
+        event.stopPropagation();
+      });
+      this.container_.appendChild(this.descriptionBox_);
```

Comment:
I think the only part that needs to be inside the `vsync.mutate` block is `this.container_.appendChild(this.descriptionBox_);` since that's the only mutation to the live DOM tree. Everything else is mutating detached nodes and has no effect on layout. The same goes for the other `build*` methods that create and modify new detached DOM nodes.

Reply:
Oh got it. Thank you!